woocommerce/woocommerce-ios
# Remove deprecated Dashboard V3 files

closes https://github.com/woocommerce/woocommerce-ios/issues/2522

# Why
Since the 1st of September, we no longer support StatsV3 for the dashboard, instead, a deprecated stats screen will be shown asking the user to upgrade to the newest version of Woo.

This PR removes all(hopefully) the code related to StatsV3 since it won't be used anymore.

# How
- Removed unused code, classes, storyboards, extensions, tests, and core data entities.
- Removed sample `jsons` that were used on tests.
- Moved a function and a struct from `StatsStore` to `StatsStoreV4`

# Testing Steps
1. Checkout `develop` branch and modify the following lines of code to be able to use the old stats in the app.
On `DashboardUIFactory.swift` on line `119`
```swift
dateComponents.month = 10
```

On `StatsVersionCoordinator.swift` on line `45`
```swift
let nextVersion: StatsVersion = .v3
```
2. Run the app and see that the old stats are presented and navigate through the dates to populate the CD store
<img width="405" alt="step1" src="https://user-images.githubusercontent.com/562080/92009504-3d2e7000-ed0e-11ea-88c2-dd6028024e70.png">
.

3. Reset local changes and checkout `issue/2522-remove-Dashboard-V3` branch and modify the following lines to make V3 the default version.
On `StatsVersionCoordinator.swift` on line `45`
```swift
let nextVersion: StatsVersion = .v3
```

4. Run the app and see that the deprecated screen is shown
<img width="400" alt="step2" src="https://user-images.githubusercontent.com/562080/92009788-a3b38e00-ed0e-11ea-85fd-a7d29ca46c96.png">

4. Reset changes and run the app again to see the stats v4 dashboard
<img width="404" alt="step3" src="https://user-images.githubusercontent.com/562080/92009940-d1003c00-ed0e-11ea-8ab4-4d1d236ff4cb.png">

5. Delete the app or it will crash when you resume your normal dev work on your branch because of using an outdated core data model version.
